client/my-sites/checkout/modal/async.tsx

```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react';
+import AsyncLoad from 'calypso/components/async-load';
+import { KEY_PRODUCTS } from './constants';
+import type { Props } from '.';
+
+const useProducts = () => {
+	const { search } = window.location;
+	const products = useMemo( () => new URLSearchParams( search ).get( KEY_PRODUCTS ), [ search ] );
```

Comment:
Super nit: can we just `return useMemo(...` directly?

The function name is already `useProducts`, so we know that we are returning "products" here, so (I feel) an intermediate variable `products` here is kind of redundant 😄

Reply:
I'm okay with both but I prefer to give it an explicit name. I'll keep it this way except you have a strong preference.

Reply:
No strong preference :)